### ISSUE TITLE
SQLi using scientific notation not detected at PL1

### ISSUE THREAD

**Issue by dune73** [coreruleset/coreruleset#2318]
### Description

New SQLi attack using scientific notation of numbers:

https://blog.h3xstream.com/2021/10/bypassing-modsecurity-waf.html

Author thinks libinjection should catch it, maybe better to catch it with regexp rules too.

We catch it just fine at PL2, but PL1 would be sweet.

```
curl localhost -d "foo=1.e(ascii 1.e(substring(1.e(select password from users limit 1 1.e,1 1.e) 1.e,1 1.e,1 1.e)1.e)1.e) = 70 or'1'='2"
```

### Your Environment

* CRS version (e.g., v3.2.0): CRS 3.4/dev
* Paranoia level setting: PL1
* ModSecurity version (e.g., 2.9.3): 2.9.4
* Web Server and version (e.g., apache 2.4.41): Apache
* Operating System and version: Ubuntu

### Confirmation

[X] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by fzipi** [comment]
Found this around: https://github.com/mindhack03d/WAF-Rule-Scientific-Notation

**Comment by theseion** [comment]
Waiting for @lifeforms to provide test set of FP's.

**Comment by dune73** [comment]
@lifeforms could you check this out please?

**Comment by dune73** [comment]
@emphazer volunteered to work on this, but he can't without the FPs promised by @lifeforms.

**Comment by lifeforms** [comment]
I totally forgot about this, sorry. Will have a look this week.

**Comment by lifeforms** [comment]
This is an interesting bypass. According to the blog, you can insert `1.e(...)` or another number like `2.e(...)`.

Fortunately, MySQL does not accept a space between `e` and `(`:

```
mysql> select 10.e(1);
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.00 sec)

mysql> select 10.e (1);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '10.e (1)' at line 1
```

The fact that `e(` should be next to each other, allows us to write a rule that hopefully will not cause many false positives.

If I'm right, we can catch this by looking for `\d\.e(` and I'm pretty positive about it.

In the comments I was asked for FP's, but I totally forgot what input I need to deliver. Does someone remember and can request me something specific?

**Comment by theseion** [comment]
@lifeforms here's where we were talking about it: https://owasp.slack.com/archives/CBKGH8A5P/p1640034126356900. You said something about having a dump from Reddit.

**Comment by fzipi** [comment]
@lifeforms @theseion We need to fix this asap. With all the changes in the SQL detection we are doing, this still holds?

@karelorigin can you help taking a look? I know you were rewriting data files, so this might be in the top of your head by now...

**Comment by fzipi** [comment]
@dune73 @lifeforms More than one year. Is this relevant, are we going to do something, or we close it?

**Comment by dune73** [comment]
OK. We're catching this at PL2 as is. Catching it at PL1 would have been cool, but there is no pressing need (as we catch it at PL2).

We're unsure of FPs and any systematic FP tests will only be available after CRS4. I am closing this now to get rid of the deadlock. And if this pops up again in the future we have the necessary discussion and the link to a PoC rule here.
